### PR TITLE
[libc] Implement roundeven C23 math functions

### PR DESCRIPTION
**Pull request by vinayakdsci**
Implements the functions `roundeven()`, `roundevenf()`, `roundevenl()` from the roundeven family of functions introduced in C23. Also implements `roundevenf128()`.

### CODE REVIEW

**Comment by vinayakdsci** [comment]
cc @lntue

**Review by nickdesaulniers — COMMENTED** [review]
Please update libc/docs/c23.rst too!

**Comment by vinayakdsci** [comment]
> Please update libc/docs/c23.rst too!

I have to remove the asterisk in front of the function name, right?

**Comment by nickdesaulniers** [comment]
> > Please update libc/docs/c23.rst too!
> 
> I have to remove the asterisk in front of the function name, right? There is an asterisk in front of all the function names.

Just add `|check|` after the name. The asterisk is because I was too lazy to spell out all of the variants. This will all get cleaned up imminently anyways (once #87682 lands, we're going to start rewriting some documentation).

File: libc/config/linux/x86_64/entrypoints.txt [hunk 1]
```diff
@@ -494,6 +494,10 @@ set(TARGET_LIBM_ENTRYPOINTS
     libc.src.math.round
     libc.src.math.roundf
     libc.src.math.roundl
+    libc.src.math.roundeven
+    libc.src.math.roundevenf
+    libc.src.math.roundevenl
+    libc.src.math.roundevenf128
```
Comment (lntue) [inline libc/config/linux/x86_64/entrypoints.txt:500]: Move `f128` variant down with other `f128` functions where they are guarded.

Comment (vinayakdsci) [reply libc/config/linux/x86_64/entrypoints.txt]: Sorry! I somehow didn't notice that. Fixed now.

File: libc/utils/MPFRWrapper/MPFRUtils.cpp [hunk 2]
```diff
@@ -351,6 +351,12 @@ class MPFRNumber {
     return result;
   }
 
+  MPFRNumber roundeven() const {
+    MPFRNumber result(*this);
+    mpfr_roundeven(result.value, value);
```
Comment (lntue) [inline libc/utils/MPFRWrapper/MPFRUtils.cpp:356]: Can you double check which version of MPFR adding `mpfr_roundeven` function?

Comment (lntue) [reply libc/utils/MPFRWrapper/MPFRUtils.cpp:356]: Can you add the version guard, and implement a replacement for `mpfr_roundeven` when it's not available, similar to https://github.com/llvm/llvm-project/blob/843cc474faefad1d639f4c44c1cf3ad7dbda76c8/libc/utils/MPFRWrapper/MPFRUtils.cpp#L244

Comment (vinayakdsci) [reply libc/utils/MPFRWrapper/MPFRUtils.cpp:358]: I implemented the replacement, but I am not sure if I should simply write `MPFR_VERSION_MAJOR >= 4`, or also specify the minor version. I went with the latter, but will change if required.

Comment (lntue) [reply libc/utils/MPFRWrapper/MPFRUtils.cpp:358]: Just check `MPFR_VERSION_MAJOR >= 4` should be enough.

Comment (vinayakdsci) [reply libc/utils/MPFRWrapper/MPFRUtils.cpp:357]: I also factored out the common code from the version guard.

**Review by lntue — APPROVED** [review]
Thanks for adding this!  I'll merge the patch later in the morning so that I can watch the bots.

**Comment by vinayakdsci** [comment]
> Thanks for adding this! I'll merge the patch later in the morning so that I can watch the bots.

Thanks a lot!

**Review by lntue — APPROVED** [review]

**Pull request merged**